cli/src/cmd/cast/storage.rs

```diff
@@ -25,55 +29,38 @@ const MIN_SOLC: Version = Version::new(0, 6, 5);
 /// CLI arguments for `cast storage`.
 #[derive(Debug, Clone, Parser)]
 pub struct StorageArgs {
-    // Storage
-    #[clap(
-        help = "The contract address.", 
-        value_name = "ADDRESS",
-        value_parser = NameOrAddress::from_str
-    )]
+    /// The contract address
+    #[clap(value_parser = NameOrAddress::from_str)]
     address: NameOrAddress,
-    #[clap(
-        help = "The storage slot number (hex or decimal)",
-        value_parser = parse_slot,
-        value_name = "SLOT"
-    )]
+
+    /// The storage slot number
+    #[clap(value_parser = parse_slot)]
     slot: Option<H256>,
-    #[clap(long, env = "ETH_RPC_URL", value_name = "URL")]
-    rpc_url: Option<String>,
-    #[clap(
-        long,
-        short = 'B',
-        help = "The block height you want to query at.",
-        long_help = "The block height you want to query at. Can also be the tags earliest, latest, or pending.",
-        value_name = "BLOCK"
-    )]
+
+    /// The block height you want to query at
+    ///
+    /// Can also be the tags earliest, latest, or pending
+    #[clap(long, short = 'B')]
     block: Option<BlockId>,
 
-    // Etherscan
-    #[clap(long, short, env = "ETHERSCAN_API_KEY", help = "etherscan API key", value_name = "KEY")]
-    etherscan_api_key: Option<String>,
-    #[clap(
-        long,
-        visible_alias = "chain-id",
-        env = "CHAIN",
-        help = "The chain ID the contract is deployed to.",
-        default_value = "mainnet",
-        value_name = "CHAIN"
-    )]
-    chain: Chain,
-
-    // Forge
+    #[clap(flatten)]
+    rpc: RpcOpts,
+
+    #[clap(flatten)]
+    etherscan: EtherscanOpts,
+
     #[clap(flatten)]
     build: build::CoreBuildArgs,
 }
 
 impl StorageArgs {
     pub async fn run(self) -> Result<()> {
-        let Self { address, block, build, rpc_url, slot, chain, etherscan_api_key } = self;
+        let Self { address, slot, block, rpc, etherscan, build } = self;
 
-        let rpc_url = try_consume_config_rpc_url(rpc_url)?;
-        let provider = try_get_http_provider(rpc_url)?;
+        let figment = Figment::from(Config::figment()).merge(etherscan).merge(rpc);
+        let config = Config::from_provider(figment);
```

Comment:
should we also merge the build args here?

worth using the figment macros+Provider impl for this type?